ethstorage/p2p/protocol/syncclient.go

```diff
@@ -234,41 +229,48 @@ func (s *SyncClient) setSyncDone() {
 	if s.mux != nil {
 		s.mux.Send(EthStorageSyncDone{DoneType: AllShardDone})
 	}
-	log.Info("Sync done", "timeUsed", s.totalSecondsUsed)
+	log.Info("Sync done")
 }
 
 func (s *SyncClient) loadSyncStatus() {
-	// Start a fresh sync for retrieval.
-	s.blobsSynced, s.syncedBytes = 0, 0
-	s.emptyBlobsToFill, s.emptyBlobsFilled = 0, 0
-	s.totalSecondsUsed = 0
 	var progress SyncProgress
 
-	if status, _ := s.db.Get(syncStatusKey); status != nil {
+	if status, _ := s.db.Get(syncTasksKey); status != nil {
 		if err := json.Unmarshal(status, &progress); err != nil {
 			log.Error("Failed to decode storage sync status", "err", err)
 		} else {
-			for _, task := range progress.Tasks {
-				log.Debug("Load sync subTask", "contract", task.Contract.Hex(),
-					"shard", task.ShardId, "count", len(task.SubTasks))
-				task.healTask = &healTask{
+			for _, t := range progress.Tasks {
+				// TODO if t.State is nil, that mean the status is marshal by old state,
+				// set process value to SyncState to make it compatible.
+				// it can be removed after public test done.
+				if t.State == nil {
+					t.State = &SyncState{
```

Comment:
what if a user is syncing, the blobsSynced will be reset to zero and the report process would still be correct?